### DISCUSSION TITLE
Typescript problem

### DISCUSSION THREAD

**Discussion by and75** [fastify/fastify#5305] [Q&A]
Good morning my name is Andrea I am a full stack develloper, at the moment I am using Fastify for a microservice. But I encounter many problems with the Typescript integration. Right now I have a problem with the creation of an asynchronous function in a decorate.   Here is my code:  `export default fp<SupportPluginOptions>(async (fastify, opts) => {

```typescript
  fastify.register(jwt, {
    secret: "sAiren2024-YUHKJ"
  })

  fastify.decorate("authenticate", async function (request:FastifyRequest, reply:FastifyReply) {
    try {
      await request.jwtVerify()
    } catch (err) {
      reply.send(err)
    }
  })
})

// When using .decorate you have to specify added properties for Typescript
declare module 'fastify' {
  export interface FastifyInstance {
    authenticate(): any ;
  }
}
```

I have this error :  

```
No overload matches this call.
  Overload 1 of 4, '(property: string | symbol, value: null): FastifyInstance<RawServerDefault, IncomingMessage, ServerResponse<IncomingMessage>, FastifyBaseLogger, FastifyTypeProviderDefault>', gave the following error.
    Argument of type '(request: FastifyRequest<RouteGenericInterface, RawServerDefault, IncomingMessage, FastifySchema, FastifyTypeProviderDefault, unknown, FastifyBaseLogger, ResolveFastifyRequestType<...>>, reply: FastifyReply<...>) => Promise<...>' is not assignable to parameter of type 'null'.
```

**Comment by mcollina** [comment]
Thanks for reporting!

Can you provide steps to reproduce? We often need a [reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), e.g. some code that allows someone else to recreate your problem by just copying and pasting it. If it involves more than a couple of different file, create a new repository on GitHub and add a link to that.

**Comment by swastikmishra** [comment]
```
server.register(fastifyJwt, {
  secret: process.env.JWT_SECRET || "defaultSecret",
});
```
`secret` expects a string, so adding || "" fixes the typescript issue.

**Reply by Uzlopak** [reply]
You could actually create an environment.d.ts and there you define it like this. But you should kind of check at startup of your service that JWT_SECRET is actually not an empty string

```ts
declare global {
  namespace NodeJS {
    interface ProcessEnv {
      JWT_SECRET: string
    }
  }
}
```
